cvmfs/download.cc

```diff
@@ -231,9 +231,13 @@ static size_t CallbackCurlHeader(void *ptr, size_t size, size_t nmemb,
   if ((info->destination == kDestinationMem) &&
       HasPrefix(header_line, "CONTENT-LENGTH:", true))
   {
-    char *tmp = reinterpret_cast<char *>(alloca(num_bytes+1));
     uint64_t length = 0;
-    sscanf(header_line.c_str(), "%s %"PRIu64, tmp, &length);
+    size_t pos = header_line.find(":") + 1;
+    assert(header_line.length() > pos);
+    if (header_line[pos] == ' ') {
+      ++pos;
+    }
+    sscanf(&header_line[pos], "%"PRIu64, &length);
```

Comment:
It's not clear. One could think it will most probably try to get the next 64 bits of a piece of memory not reserved by the program, which can potentially make it crash. However, when reading the documentation (http://www.cplusplus.com/reference/cstdio/sscanf/) it claims it returns EOF on failure when reading.

Shall the size be checked in advance?